tests/test/mod.rs

```diff
@@ -333,6 +360,53 @@ fn solve_goal(program_text: &str, goals: Vec<(&str, SolverChoice, TestGoal)>, co
     })
 }
 
+fn solve_aggregated(
+    db: &mut ChalkDatabase,
+    program: Arc<Program>,
+    goal_text: &str,
+    choices: &[SolverChoice],
+    expected: Expect,
+) {
+    let mut solutions = vec![];
+
+    for solver_choice in choices.iter().copied() {
+        if db.solver_choice() != solver_choice {
+            db.set_solver_choice(solver_choice);
+        }
+
+        chalk_integration::tls::set_current_program(&program, || {
+            println!("----------------------------------------------------------------------");
+            println!("goal {}", goal_text);
+            assert!(goal_text.starts_with("{"));
+            assert!(goal_text.ends_with("}"));
+            let goal = lower_goal(
+                &*chalk_parse::parse_goal(&goal_text[1..goal_text.len() - 1]).unwrap(),
+                &*program,
+            )
+            .unwrap();
+
+            println!("using solver: {:?}", solver_choice);
+            let peeled_goal = goal.into_peeled_goal(db.interner());
+            let result = db.solve(&peeled_goal);
+            solutions.push(format_solution(result, db.interner()));
+        });
+    }
+
+    let (head, tail) = solutions
+        .split_first()
+        .expect("Test requires at least one solver");
+    for (i, other) in tail.iter().enumerate() {
+        println!(
+            "\ncomparing solvers:\n\tleft: {:?}\n\tright: {:?}\n",
+            &choices[0],
+            &choices[i + 1]
+        );
+        assert_same(head, other);
```

Comment:
`assert_same` strips whitespace, uses prefix matching, and has a different diff output than `Expect::assert_eq`, so the error that occurs when two solvers don't match looks quite different from the one that occurs when the expected output is wrong. `Expect::assert_eq` does not work without something to bless, though it could be made to with rust-analyzer/expect-test#22. If this is merged, I expect that uses of `assert_same ` will slowly be removed in favor of `Expect`, so I don't think we need to resolve this right now. Others may disagree, however.